src/imitation/scripts/common/rl.py

```diff
@@ -84,13 +117,19 @@ def make_rl_algo(
         ), "set 'n_steps' at top-level using 'batch_size'"
         rl_kwargs["n_steps"] = batch_size // venv.num_envs
     elif issubclass(rl_cls, off_policy_algorithm.OffPolicyAlgorithm):
-        assert "batch_size" not in rl_kwargs, "set 'batch_size' at top-level"
+        if "batch_size" not in rl_kwargs or rl_kwargs["batch_size"] is not None:
+            raise ValueError("set 'batch_size' at top-level")
         rl_kwargs["batch_size"] = batch_size
     else:
         raise TypeError(f"Unsupported RL algorithm '{rl_cls}'")
     rl_algo = rl_cls(
         policy=train["policy_cls"],
-        policy_kwargs=train["policy_kwargs"],
+        # Note(yawen): Here we make a copy of policy_kwargs as a temporary workaround
```

Comment:
This is a good point. What's temporary about it -- do you have an alternative solution in mind? If SB3 is going to go around mutating our configs I think this is the right thing to do.

I'd say something like:
```
# Note(yawen): Copy `policy_kwargs` as SB3 may mutate the config we pass.
# In particular, policy_kwargs["use_sde"] may be changed in rl_cls.__init__() for
# certain algorithms, such as Soft Actor Critic. See:
# https://github.com/DLR-RM/stable-baselines3/blob/30772aa9f53a4cf61571ee90046cdc454c1b11d7/stable_baselines3/common/off_policy_algorithm.py#L145
```

Reply:
I don't have an alternative solution. There is [another way](https://github.com/IDSIA/sacred/issues/502) around: which is to set `SETTINGS.CONFIG.READ_ONLY_CONFIG = False`, but this is even less desirable because it would have effects globally.
```python
from sacred import SETTINGS
SETTINGS.CONFIG.READ_ONLY_CONFIG = False
```

Reply:
Perhaps we should also add a warning here?

Reply:
Isn't the sacred warning here correct, though? Unless I'm missing something we don't want config objects to be mutated. So turning off the warning seems worse than copying it (which actually stops it from being mutated).

Reply:
Mutating config objects with `SETTINGS.CONFIG.READ_ONLY_CONFIG = True` (which is the default) actually raises an exception instead of a warning. I think we agree that copying is better than setting `SETTINGS.CONFIG.READ_ONLY_CONFIG = False`. We can go ahead with the current approach of copying the config if you agree.